src/wasapi.c

```diff
@@ -666,40 +666,48 @@ static int refresh_devices(struct SoundIoPrivate *si) {
     if (FAILED(hr = IMMDeviceEnumerator_GetDefaultAudioEndpoint(siw->device_enumerator, eRender,
                     eMultimedia, &rd.default_render_device)))
     {
-        deinit_refresh_devices(&rd);
-        return SoundIoErrorOpeningDevice;
-    }
-    if (rd.lpwstr) {
-        CoTaskMemFree(rd.lpwstr);
-        rd.lpwstr = NULL;
-    }
-    if (FAILED(hr = IMMDevice_GetId(rd.default_render_device, &rd.lpwstr))) {
-        deinit_refresh_devices(&rd);
-        return SoundIoErrorOpeningDevice;
+        if(hr != E_NOTFOUND) {
+            deinit_refresh_devices(&rd);
+            return SoundIoErrorOpeningDevice;
+        }
     }
-    if ((err = from_lpwstr(rd.lpwstr, &rd.default_render_id, &rd.default_render_id_len))) {
-        deinit_refresh_devices(&rd);
-        return err;
+    if(rd.default_render_device) {
```

Comment:
The docs say that this is guaranteed to be null in the event that there isn't one but it might be cleared to have the same check as above (that hr != E_NOTFOUND).